pkg/pushsync/pushsync.go

```diff
@@ -314,18 +268,22 @@ func (ps *PushSync) PushChunkToClosest(ctx context.Context, ch swarm.Chunk) (r *
 			}
 		}
 
-		receipt, err := ps.receiveReceipt(ctx, r)
-		if err != nil {
+		var receipt pb.Receipt
+		cctx, cancel := context.WithTimeout(ctx, timeToWaitForReceipt)
+		defer cancel()
+		if err := r.ReadMsgWithContext(cctx, &receipt); err != nil {
 			ps.metrics.TotalErrors.Inc()
 			_ = streamer.Reset()
```

Comment:
here we can  just release reserved price

Reply:
it is done in the defer call